src/ui/DynamicFacetManager/DynamicFacetManager.ts

```diff
@@ -232,13 +232,26 @@ export class DynamicFacetManager extends Component {
 
     const [collapsableFacets, uncollapsableFacets] = partition(this.facetsWithDisplayedValues, facet => facet.options.enableCollapse);
     const [facetsWithActiveValues, remainingFacets] = partition(collapsableFacets, facet => facet.hasActiveValues);
-    const indexOfFirstFacetToCollapse =
+    let indexOfFirstFacetToCollapse =
       this.options.maximumNumberOfExpandedFacets - uncollapsableFacets.length - facetsWithActiveValues.length;
 
     facetsWithActiveValues.forEach(dynamicFacet => dynamicFacet.expand());
 
     remainingFacets.forEach((dynamicFacet, index) => {
-      index < indexOfFirstFacetToCollapse ? dynamicFacet.expand() : dynamicFacet.collapse();
```

Comment:
Hmm, I really liked this old logic. Isn't the new behaviour simply adding a check on the `collapseByDefault` option? e.g.

```
const canExpand = !dynamicFacet.options.collapsedByDefault;
index < indexOfFirstFacetToCollapse ? canExpand && dynamicFacet.expand() : dynamicFacet.collapse();
```

I think it's nicer than needing to dynamically increment the index.

Reply:
We increment it because we need to expand another facet instead of this one.

Reply:
Ok, I see, would switching to a counter model be any simpler at this point rather than using a moving index?
Each facet gets checked for whether it can be expanded/collpased. If we can expand, we decrement the counter? When the counter is 0, we can no longer expand.

The equation on line 235 combined now with a moving index make this piece of code complex to reason about, so I wonder if it is the simplest solution?